egs/librispeech/ASR/conformer_ctc/gigaspeech_datamodule.py

```diff
@@ -0,0 +1,403 @@
+            '--on-the-fly-feats',
+            dest="giga_on_the_fly_feats",
+            type=str2bool,
+            default=False,
+            help='When enabled, use on-the-fly cut mixing and feature extraction. '
+                 'Will drop existing precomputed feature manifests if available.'
+        )
+        group.add_argument(
+            '--shuffle',
+            dest="giga_shuffle",
+            type=str2bool,
+            default=True,
+            help='When enabled (=default), the examples will be shuffled for each epoch.'
+            )
+        group.add_argument(
+            '--check-cuts',
+            dest="giga_check_cuts",
+            type=str2bool,
+            default=True,
+            help='When enabled (=default), we will iterate over the whole training cut set '
+                 'to validate it. It should be disabled when using Apache Arrow manifests '
+                 'to avoid an excessive starting time of the script with datasets>1000h.'
+            )
+
+        # GigaSpeech specific arguments
+        group.add_argument(
+            "--subset",
+            dest="giga_subset",
+            type=str,
+            default="XS",
+            help="Select the GigaSpeech subset (XS|S|M|L|XL)",
+        )
+        group.add_argument(
+            "--context-window",
+            dest="giga_context_window",
+            type=float,
+            default=0.0,
+            help="Training cut duration in seconds. "
+                 "Use 0 to train on supervision segments without acoustic context, with variable cut lengths; "
+                 "number larger than zero will create multi-supervisions cuts with actual acoustic context. ",
+        )
+        group.add_argument(
+            "--context-direction",
+            dest="giga_context_direction",
+            type=str,
+            default="center",
+            help="If context-window is 0, does nothing. "
+                 "If it's larger than 0, determines in which direction (relative to the supervision) "
+                 "to seek for extra acoustic context. Available values: (left|right|center|random).",
+        )
+        group.add_argument(
+            '--use-context-for-test',
+            dest="giga_use_context_for_text",
+            type=str2bool,
+            default=False,
+            help='Should we read cuts with acoustic context or without it. '
+                 '(note: for now, they may contain duplicated segments)'
+        )
+        group.add_argument(
+            '--small-dev',
+            dest="giga_small_dev",
+            type=str2bool,
+            default=False,
+            help='Should we use only 1000 utterances for dev (speeds up training)'
+        )
+
+    def validate_args(self):
+        if self.args.giga_subset in ['L', 'XL']:
+            assert (
+                self.args.giga_shuffle == False
+            ), "For GigaSpeech L/XL, you must use --shuffle 0 to avoid eagerly reading pyarrow manifests."
```

Comment:
As mentioned in other comment -- this file also seems to come from an older version of the recipe.

Reply:
This part is the same with snowfall master.  https://github.com/k2-fsa/snowfall/blob/master/egs/gigaspeech/asr/simple_v1/asr_datamodule.py#L183

Shall I removed the check of shuffle==False and bucketing_sampler==False? (I already removed the check of on_the_fly_feats)

Reply:
You can remove the shuffle==False check (I added streaming pseudo shuffling in Lhotse) but I’m not sure about the BucketingSampler — I think I still haven’t added support for very large data for it (it will take long time, a few minutes maybe, to start the script if you turn it on). I can try to add it soon.